### PR TITLE
ref(issues): Refactor group event details

### PR DESCRIPTION
**Pull request by lynnagara**
Refactor group event details to make it more usuable for the
environments view. GroupDetails always passes project to child
routes via props since it will no longer be accessible via context.

### CODE REVIEW

File: src/sentry/static/sentry/app/views/groupDetails/shared/groupDetails.jsx [hunk 1]
```diff
@@ -21,7 +21,11 @@ const GroupDetails = createReactClass({
   displayName: 'GroupDetails',
 
   propTypes: {
+<<<<<<< HEAD
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/groupDetails/shared/groupDetails.jsx]: Some merge markers made their way in here

Comment (lynnagara) [reply src/sentry/static/sentry/app/views/groupDetails/shared/groupDetails.jsx]: 😱 thanks!

File: src/sentry/static/sentry/app/views/groupDetails/shared/utils.jsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ * Fetches group data and mark as seen
+ *
+ * @param {String} orgId organization slug
+ * @param {String} projectId project slug
+ * @param {String} groupId groupId
+ * @param {String} eventId eventId or "latest" or "oldest"
+ * @returns {Promise<Object>}
+ */
+export function fetchGroupEventAndMarkSeen(orgId, projectId, groupId, eventId) {
+  const api = new Client();
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/groupDetails/shared/utils.jsx]: Should the containing component pass the client in? That allows the component to cancel any pending API requests should it be unmounted.

**Review by markstory — APPROVED** [review]

File: src/sentry/static/sentry/app/views/groupDetails/shared/utils.jsx (hunk 2 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/views/groupDetails/shared/utils.jsx]: There's a `withApi` HoC

**Pull request merged**